pgmpy/factors/Factor.py

```diff
@@ -443,8 +443,11 @@ def _str(self, html=False):
         # to increment it
         # by 1.
         def fun(b, index=len(self.cardinality)-1):
+            #print ('---->', self.cardinality, index, b)
```

Comment:
@ashwch When submitting a PR make sure that you remove the debug statements completely instead of commenting them.